website/content/en/docs/contribution-guidelines/releasing.md

```diff
@@ -60,38 +67,31 @@ We will use the `v1.3.0` release version in this example.
    by running the
    [deploy-manual](https://github.com/operator-framework/operator-sdk/actions/workflows/deploy-manual.yml)
    GitHub action.  After the image is built, check the security scan
-   results under `Child Manifests`.
-1. A release branch must be created. If you have the proper permissions,
-   you can do this by running the following, assuming the upstream SDK
+   results under `Child Manifests`. Merge the resulting pull requests to update the links.
+1. Make sure that the list of supported OLM versions stated in the [Overview][overview] section of SDK docs is updated. If a new version of OLM needs to be officially supported, follow the steps in [updating OLM bindata](#updating-olm-bindata) section.
+1. Lock down the `master` branch to prevent further commits before the release completes:
+  1. Go to `Settings -> Branches` in the SDK repo.
+  1. Under `Branch protection rules`, click `Edit` on the `master` branch rule.
+  1. In section `Protect matching branches` of the `Rule settings` box, increase the number of required approving reviewers to 6.
+
+### 1. Create and push a release commit
+
+   A release branch must be created. Note that the release branch is labeled `1.3.x` while the tag
+   is labeled `1.3.0`. Future patch releases will have their own tags but be based on the same release
+   branch. If you have the proper permissions, you can do this by running the following, assuming the upstream SDK
    is the `upstream` remote repo:
   ```sh
   git checkout master
   git pull
   git checkout -b v1.3.x
   git push -u upstream v1.3.x
   ```
-1. Make sure that the list of supported OLM versions stated in the [Overview][overview] section of SDK docs is updated. If a new version of OLM needs to be officially supported, follow the steps in [updating OLM bindata](#updating-olm-bindata) section.
-1. Create and merge a commit that updates the top-level [Makefile] variable `IMAGE_VERSION`
+Update the top-level [Makefile] variable `IMAGE_VERSION`
 to the upcoming release tag `v1.3.0`. This variable ensures sample projects have been tagged
 correctly prior to the release commit.
   ```sh
   sed -i -E 's/(IMAGE_VERSION = ).+/\1v1\.3\.0/g' Makefile
   ```
-1. Lock down the `master` branch to prevent further commits before the release completes:
-  1. Go to `Settings -> Branches` in the SDK repo.
-  1. Under `Branch protection rules`, click `Edit` on the `master` branch rule.
-  1. In section `Protect matching branches` of the `Rule settings` box, increase the number of required approving reviewers to 6.
-
-### 1. Create and push a release commit
-
-Create a new branch to push the release commit:
-
-```sh
-export RELEASE_VERSION=v1.3.0
-git checkout master
-git pull
-git checkout -b release-$RELEASE_VERSION
-```
```

Comment:
Why are we removing this? I think it's confusing to work in `v1.3.x` in `origin` which is my fork. It can easily be confused with the actual release branch we want. Having a `release-v1.3.0` branch in `origin` makes much more sense to me since it's explicitly the current release I'm creating.